java/integTest/src/test/java/glide/SharedCommandTests.java

```diff
@@ -4230,4 +4233,48 @@ public void lset(BaseClient client) {
         String[] updatedList2 = client.lrange(key, 0, -1).get();
         assertArrayEquals(updatedList2, expectedList2);
     }
+
+    @SneakyThrows
+    @ParameterizedTest(autoCloseArguments = false)
+    @MethodSource("getClients")
+    public void lmove(BaseClient client) {
+        assumeTrue(REDIS_VERSION.isGreaterThanOrEqualTo("6.2.0"), "This feature added in redis 6.2.0");
+        // setup
+        String key1 = "{key}-1" + UUID.randomUUID();
+        String key2 = "{key}-2" + UUID.randomUUID();
+        String nonExistingKey = "{key}-3" + UUID.randomUUID();
+        String nonListKey = "{key}-4" + UUID.randomUUID();
+        String[] lpushArgs1 = {"four", "three", "two", "one"};
+        String[] lpushArgs2 = {"six", "five", "four"};
+
+        // source does not exist or is empty
+        assertNull(client.lmove(key1, key2, ListDirection.LEFT, ListDirection.RIGHT).get());
+
+        // only source exists, only source elements gets popped, nothing gets added
```

Comment:
```suggestion
        // only source exists, creates a list at nonExistingKey
```
I think you will find that nonExistingKey contains a list now...